app/controllers/shares_controller.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+# Controller for Shares
+class SharesController < ApplicationController
+  before_action :set_work
+
+  def new
+    authorize! @work, with: SharePolicy
+
+    @form = WorkShareForm.new(shares_attributes: @work.shares.map do |share|
+      { sunetid: share.user.sunetid,
+        name: share.user.name,
+        permission: share.permission }
+    end)
+  end
+
+  def create
+    authorize! @work, with: SharePolicy
+
+    form = WorkShareForm.new(**share_params)
+    ShareUpdater.call(work: @work, form:)
+
+    redirect_to work_path(@work)
+  end
+
+  private
+
+  def set_work
+    @work = Work.find_by!(druid: params[:work_druid])
+  end
+
+  def share_params
+    if params.include?(:work_share)
+      params.expect(work_share: [WorkShareForm.nested_attributes])
+    else
+      { shares_attributes: [] }
+    end
+  end
+
+  # Updates the shares of a work based on the provided form.
+  class ShareUpdater
```

Comment:
can we move this elsewhere so we don't have a class definition embedded within the controller?  like maybe into the service folder ?

Reply:
We could, but it is only used here. This keeps the services directory slim and puts it near the code in which it is used.

Reply:
Understood.  We don't have a shared practice on this since we have other cases of a class being used in only one place but defined in it's own file with a filename matching the class, which is what I'd typically expect.  It's not a big deal, but my first instinct when seeing a reference to a class is to go looking for it's definition in the typically expected spots and not to scroll down in the same file (particularly in a controller).